src/content/en/updates/2020/11/cvd.md

```diff
@@ -0,0 +1,461 @@
+      filtered.style.filter = value;
+    }
+  });
+</script>
+{% endframebox %}
+
+If we wanted to, we could build our DevTools feature as follows: when the user emulates a vision deficiency in the DevTools UI, we inject the SVG filter into the inspected document, and then we apply the filter style on the root element. However, there are several problems with that approach:
+
+The page might already have a filter on its root element, which our code might then override.
+The page might already have an element with `id="deuteranopia"`, clashing with our filter definition.
+The page might rely on a certain DOM structure, and by inserting the `<svg>` into the DOM we might violate these assumptions.
+
+Edge cases aside, the main problem with this approach is that *we’d be making programmatically observable changes to the page*. If a DevTools user inspects the DOM, they might suddenly see an `<svg>` element they never added, or a CSS `filter` they never wrote. That would be confusing! To implement this functionality in DevTools, we need a solution that doesn’t have these drawbacks.
+
+Let’s see how we can make this less intrusive. There’s two parts to this solution that we need to hide: 1) the CSS style with the `filter` property, and 2) the SVG filter definition, which is currently part of the DOM.
+
+```html
+<!-- Part 1: the CSS style with the filter property -->
+<style>
+  :root {
+    filter: url(#deuteranopia);
+  }
+</style>
+<!-- Part 2: the SVG filter definition -->
+<svg>
+  <filter id="deuteranopia">
+    <feColorMatrix values="0.367  0.861 -0.228  0.000  0.000
+                           0.280  0.673  0.047  0.000  0.000
+                          -0.012  0.043  0.969  0.000  0.000
+                           0.000  0.000  0.000  1.000  0.000">
+  </filter>
+</svg>
+```
+
+### Avoiding the in-document SVG dependency
+
+Let’s start with part 2: how can we avoid adding the SVG to the DOM? One idea is to move it to a separate SVG file. We can copy the `<svg>…</svg>` from the above HTML and save it as `filter.svg` — but we need to make some changes first! Inline SVG in HTML follows the HTML parsing rules. That means you can get away with things like [omitting quotes around attribute values in some cases](https://mathiasbynens.be/notes/unquoted-attribute-values). However, SVG in separate files is supposed to be valid XML — and XML parsing is way more strict than HTML. Here’s our SVG-in-HTML snippet again:
+
+```html
+<svg>
+  <filter id="deuteranopia">
+    <feColorMatrix values="0.367  0.861 -0.228  0.000  0.000
+                           0.280  0.673  0.047  0.000  0.000
+                          -0.012  0.043  0.969  0.000  0.000
+                           0.000  0.000  0.000  1.000  0.000">
+  </filter>
+</svg>
+```
+
+To make this valid standalone SVG (and thus XML), we need to make two changes. Can you guess which?
+
+```xml
+<svg xmlns="http://www.w3.org/2000/svg">
```

Comment:
I guess there's no way to highlight specific lines, or even characters, in WebFu code snippets?

Reply:
Possible, see the escape version of the code with `<pre>` tag and code highlight with `<strong>`.